src/agents/parking_agent.py

```diff
@@ -1,33 +1,151 @@
 import spade
+import string
+import json
+import random
 from spade.agent import Agent
-from spade.behaviour import CyclicBehaviour, OneShotBehaviour
+from spade.behaviour import CyclicBehaviour
 from spade.message import Message
 from spade.template import Template
 
+from constants import DEFAULT_HOST, PARKING_AGENT_MESSAGE_TIMEOUT
+from messages.check_parking import CheckParking
+from messages.parking_availability import ParkingAvailable
+from messages.reservation_request import RequestReservation
+from messages.response import ReservationResponse
+from messages.modify_reservation import ModifyReservation
+
 
 class ParkingAgent(Agent):
-    class CheckParkingBehaviour(CyclicBehaviour):
+    def __init__(self, jid, password, x, y, parking_spots):
+        super().__init__(jid, password)
+        self._x = x
+        self._y = y
+        self._parking_spots = parking_spots
+        self._available_parking_spots = [
+            parking_spots for _ in range(24)
+        ]  # a list with number of free/available spots at a given hour
+
+    def _prepare_check_parking_spots_template(self):
+        template = Template()
+        template.to = f"{self.jid}@{DEFAULT_HOST}"
+        template.set_metadata("performative", "inform")
+        template.set_metadata("inform", "parking-availability")
+        return template
+
+    def _prepare_make_reservation_template(self):
+        template = Template()
+        template.to = f"{self.jid}@{DEFAULT_HOST}"
+        template.set_metadata("performative", "inform")
+        template.set_metadata("inform", "make-reservation")
+        return template
+
+    def _prepare_modify_reservation_template(self):
+        template = Template()
+        template.to = f"{self.jid}@{DEFAULT_HOST}"
+        template.set_metadata("performative", "inform")
+        template.set_metadata("inform", "modify-reservation")
+        return template
+
+    class CheckParkingSpots(CyclicBehaviour):
+        """Behaviour for answering requests for parking spots"""
+
+        async def run(self):
+            msg = await self.receive(timeout=PARKING_AGENT_MESSAGE_TIMEOUT)
+            if msg:
+                check_parking_message = CheckParking(**json.loads(msg.body))
+                available_spots = self.get_available_parking_spots(
+                    check_parking_message.time_start, check_parking_message.time_stop
+                )
+                reply_body = ParkingAvailable(self.jid, 40, self._x, self._y, True if available_spots else False)
```

Comment:
Maybe let's add price as an attribute in init?

Reply:
So that we could provide it as a parameter during the Agent creation similarly as its coordinates?

Reply:
good idea